storage/src/adb/any.rs

```diff
@@ -78,18 +75,65 @@ pub struct Any<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> {
     /// are over keys that have been updated by some operation at or after this point).
     inactivity_floor_loc: u64,
 
-    /// A map from each key to the location in the log containing its most recent update. Only
-    /// contains the keys that currently have a value (that is, deleted keys are not in the map).
-    snapshot: Index<EightCap, u64>,
+    /// A snapshot of all currently active operations in the form of a map from each key to the
+    /// location in the log containing its most recent update. Only contains the keys that currently
+    /// have a value (that is, deleted keys are not in the map).
+    snapshot: Index<T, u64>,
 
     /// The number of operations that are pending commit.
     uncommitted_ops: u64,
 }
 
-impl<E: RStorage + Clock + Metrics, K: Array, V: Array, H: CHasher> Any<E, K, V, H> {
-    /// Return an MMR initialized from `cfg`. Any uncommitted operations in the log will be
+/// The result of a database `update` operation.
+pub enum UpdateResult {
+    NoOp,              // tried to set a key to its current value
```

Comment:
Since this is a public enum, we should move these explanations to be `///` above each choice.

Reply:
done